### PR TITLE
INT-91 {ADD A USER}

### PR DESCRIPTION
**Pull request by CodingWithCDJE**
* Config the routes to add a user based on the receiving request

### CODE REVIEW

**Comment by horseeyephil** [comment]
haha - I like that PR header

File: routes/api/users.js [hunk 1]
```diff
@@ -4,6 +4,21 @@ const router = express.Router();
 
 const User = require('../../database/models/User');
 
+const Project_User = require('../../database/models/Project_User');
+
+function addAuth(req, res, next) {
+  if (req.user) {
+    const title = req.user.title;
+    title.length !== 0
+      ? title.contains('Manager')
```
Comment (horseeyephil) [inline routes/api/users.js:13]: hey Chris - let's avoid "trinaries", especially for a case like the this where the results are side-effect based

File: routes/api/users.js [hunk 2]
```diff
@@ -32,6 +47,39 @@ router.get('/:id', async (req, res) => {
...
+      lastname,
+      department,
+      title,
+      avatar: 'https://avatars.io/static/default_128.jpg',
+      googleId: googleid,
+    });
+
+    const addedUser = await User.query().findById(googleid);
+
+    console.log(addedUser);
```
Comment (horseeyephil) [inline routes/api/users.js:71]: scrub logs

File: server.js [hunk 3]
```diff
@@ -82,9 +82,9 @@ passport.deserializeUser(async function (id, done) {
   return done(null, user);
 });
 
-app.get('/', (req, res) => {
-  console.log(req.user);
-});
+// app.get('/', (req, res) => {
```
Comment (horseeyephil) [inline server.js:85]: scrub extra stuff

File: routes/api/users.js (hunk 1 above)
Comment (horseeyephil) [inline routes/api/users.js:13]: line 13: I see this is a compact way to grab "manager" types - but for these auth patterns, it's better / more formal to be explicit about which roles are allowed:
try something like `if ( role) in ['engineering manager', 'project manager', 'department manager'] then` - use an explicit enum

Comment (CodingWithCDJE) [reply routes/api/users.js:13]: Understood. I thought trinaries would make the code look a bit cooler, imo then the casual if else statement.

File: routes/api/users.js (hunk 2 above)
Comment (CodingWithCDJE) [reply routes/api/users.js:71]: Alrighty 👍

File: routes/api/users.js (hunk 1 above)
Comment (CodingWithCDJE) [reply routes/api/users.js:13]: So basically you want me to have an array of roles that can match the case ?

Comment (CodingWithCDJE) [reply routes/api/users.js:13]: By that I mean the cases, in which allows you to add a user to the project.

File: server.js (hunk 3 above)
Comment (CodingWithCDJE) [reply server.js]: 👍

File: routes/api/users.js (hunk 1 above)
Comment (CodingWithCDJE) [reply routes/api/users.js]: 👍

**Pull request merged**